.gitignore

```
@@ -1,5 +1,5 @@
 .DS_Store
-/dist*
+**/dist*
```

Do we have nested `/dist`? Does this still catch the root level one?

---

@tclem my local testing shows this catches both `example/dist-newstyle` and `dist-newstyle` at the root of the repo.

---

I'm not sure this pattern is the best choice:

1. There should probably be a trailing `/` to avoid matching files.

1. `**/dist*/` is the same as `dist*/` but less clear. The `**` syntax is for when you have `**/x/y` which isn't the same as `x/y`.

1. If someone renames `dist-newstyle` to `dist-newstyle.bak` it will still be ignored, which may not be what you want.

1. You may want to add `cabal.project.local`.

I therefore suggest:

```suggestion
cabal.project.local
dist/
dist-newstyle/
```

Also worth thinking about:

1. `dist/` matches at any level, not just the top and one immediately below the top, which may be too wide. It would ignore any directory called `dist` anywhere in the hierarchy. Using both `/dist/` and `/*/dist/` together is more accurate. (And the same for `dist-newstyle/`.)

1. In general, I think it's better to have subdirectories define their own patterns rather than putting them all into a single top-level ignore file. You could add `example/.gitignore` containing `/dist/` and `/dist-newstyle/`. However, maybe this is too fussy.